core/keyboard_nav/ast_node.js

```diff
@@ -420,7 +420,7 @@ Blockly.ASTNode.prototype.findPrevForField_ = function() {
  */
 Blockly.ASTNode.prototype.navigateBetweenStacks_ = function(forward) {
   var curLocation = this.getLocation();
-  if (!(curLocation instanceof Blockly.Block)) {
+  if (curLocation.getSourceBlock) {
```

Comment:
I think the best way to fix this would be below, but if you are going for simplest possible change right now then this works. 
```
  var sourceBlock = this.getSourceBlock();
  if (!sourceBlock || !sourceBlock.workspace) {
    return null;
  }
  var curRoot = sourceBlock.getRootBlock();
```

Reply:
I can add that block of code inside of this `if block`.
We'll still need to check for the existence of the `getSourceBlock` method since it's not on the `Blockly.ASTNode` interface and some of the objects passed in may not have it defined.